### ISSUE TITLE
curiosity for efficiency of the generated multi-kernels 

### ISSUE THREAD

**Issue by preFiredman** [plaidml/plaidml#708]
Conducting some basic ops using plaidML may generated multi kernels, which entail multi temporary buffers for temp variables.  This will surely take up much memory， seems unnecessary,  in GPU, and doesn't the I/O slow down the efficiency? why don't you combine these kernels, more than ten in some cases, into less kernels?
great thanks!

**Comment by flaub** [comment]
Yes, fusion optimization passes are intended to combine kernels. Do you have an example where fusion passes aren't working? Could you share them here?

**Comment by preFiredman** [comment]
well, I mean if I write a new op using tile code, generally in my case, plaidML will generate multi kernels, at least more than one. So, why doesn't plaidML combine them into less ones?  Also, what's the difference in effiency between two kernel generation way, i.e, more kernels and less kernels, which both imply one same algorithm?

**Comment by preFiredman** [comment]
take your official document as an [example](https://vertexai-plaidml.readthedocs-hosted.com/en/latest/life-function-api.html):
```
function (T[X, Y], O[X, Y]) -> (R) {
    LO = log(O);
    Temp[x: X] = +(LO[x, y] * T[x, y]);
    R = -Temp;
}
```
this tile code will generate two kernels , I mean in some more complex cases, plaidML will generate much more kernels, why doesn't combine them into one? has compared their efficiencies?

**Comment by flaub** [comment]
Looking at tile code doesn't give you any indication as to how many kernels will be generated. The compiler uses a configuration file that is specific to each target that determines how the problem should be broken up into items that can be parallelized and scheduled on the target device.

**Comment by preFiredman** [comment]
> Looking at tile code doesn't give you any indication as to how many kernels will be generated. The compiler uses a configuration file that is specific to each target that determines how the problem should be broken up into items that can be parallelized and scheduled on the target device.

 is that to say, no matter how much kernels been generated, their total efficiency is always higher than any kind of combined kernels for a specific tile code ?
